skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,301 @@
+Starts with #=GC followed by a tag and data relating to the tag. Typically
+comes at the end of the multiple sequence alignment.
+For example:
+
+.. code-block:: none
+
+    #=GC SS_cons            CCCCCHHHHHHHHHHHHH..EEEEEEEE....EEEEEEEEEEH
+
+Example taken from [2]_.
+
+Examples
+========
+Suppose we have a Stockholm file:
+
+>>> import skbio.io
+>>> from io import StringIO
+>>> from skbio.sequence import RNA
+>>> from skbio.alignment import TabularMSA
+>>> fs = '\\n'.join([
+...         '# STOCKHOLM 1.0',
+...         '#=GF RA    Deiman BA, Kortlever RM, Pleij CW;',
+...         '#=GF RL    J Virol 1997;71:5990-5996.',
+...         'AF035635.1/619-641             UGAGUUCUCGAUCUCUAAAAUCG',
+...         'M24804.1/82-104                UGAGUUCUCUAUCUCUAAAAUCG',
+...         'J04373.1/6212-6234             UAAGUUCUCGAUCUUUAAAAUCG',
+...         'M24803.1/1-23                  UAAGUUCUCGAUCUCUAAAAUCG',
+...         '#=GC SS_cons                   .AAA....<<<<aaa....>>>>',
+...         '//'
+...     ])
+>>> fh = StringIO(fs)
+>>> msa = skbio.io.read(fh, into=TabularMSA, constructor=RNA)
+>>> msa # doctest: +NORMALIZE_WHITESPACE
+TabularMSA[RNA]
+-------------------------------------------------
+Metadata:
+'RA': '   Deiman BA, Kortlever RM, Pleij CW;'
+'RL': '   J Virol 1997;71:5990-5996.'
+Positional metadata:
+'SS_cons': <dtype: object>
+Stats:
+sequence count: 4
+position count: 23
+-------------------------------------------------
+UGAGUUCUCGAUCUCUAAAAUCG
+UGAGUUCUCUAUCUCUAAAAUCG
+UAAGUUCUCGAUCUUUAAAAUCG
+UAAGUUCUCGAUCUCUAAAAUCG
+
+References
+==========
+.. [1] https://en.wikipedia.org/wiki/Stockholm_format
+.. [2] http://sonnhammer.sbc.su.se/Stockholm.html
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from collections import OrderedDict
+
+from skbio.alignment import TabularMSA
+from skbio.sequence import Protein
+from skbio.io import create_format, StockholmFormatError
+
+stockholm = create_format('stockholm')
+
+
+@stockholm.sniffer()
+def _stockholm_sniffer(fh):
+    # Smells a Stockholm file if the first line contains 'Stockholm' and the
+    # version number.
+    line = next(fh)
+    if line[:15] == "# STOCKHOLM 1.0":
+        return True, {}
+
+    return False, {}
+
+
+@stockholm.reader(TabularMSA)
+def _stockholm_to_tabular_msa(fh, constructor=Protein):
+    # Uses OrderedDict() to make sure dna_data isn't arranged randomly
+    dna_data = OrderedDict()
+    metadata = {}
+    positional_metadata = {}
+    seqs = []
+
+    # Retrieves data from file
+    for line in fh:
```

Comment:
It looks like this can be combined with the loop below. That way you don't have to do `fh.seek(0)` and loop twice.